### ISSUE TITLE
Reactive Chart - Zooming and Scrolling (Panning)

### ISSUE THREAD

**Issue by DmitryBogomolov** [DevExpress/devextreme-reactive#1939]
# The Problem

Users cannot zoom and scroll (pan) the Chart.

# The Proposed Solution

We will provide a new `ZoomAndPan` plugin that allows users to:

- zoom in and out with touch gestures and the mouse wheel 
- zoom the chart by selecting an area with the mouse drag gesture
- pan the chart with touch gestures and the mouse drag gesture

Zooming and panning can be enabled for arguments and/or values.

## Controlled mode

```jsx
this.changeViewport = (viewport) => {
  this.setState({ viewport });
};
...
const { viewport } = this.state;
const {
  argumentStart,  // the viewport's start point on the argument scale
  argumentEnd,    // the viewport's end point on the argument scale
  valueStart,     // the viewport's start point on the value scale
  valueEnd,       // the viewport's end point on the value scale
  scaleName,      // the name of the scale for which the viewport is set
} = viewport;
...
<ZoomAndPan
  viewport={viewport}
  onViewportChange={this.changeViewport}
/>
```

## Uncontrolled mode

```jsx
const initialViewport = {
  argumentStart: 'B',
  argumentEnd: 'D',
  valueStart: 4,
  valueEnd: 9,
};
...
<ZoomAndPan
  defaultViewport={initialViewport}
/>
```

## Allow users to zoom and pan only arguments or values

```jsx
<ZoomAndPan
  interactionWithArguments="none | zoom | pan | both"
  interactionWithValues="none | zoom | pan | both"
/>
```

## Zoom the chart by selecting an area

```jsx
<ZoomAndPan
  zoomRegionKey="shift"
  dragBoxComponent={/* component that renders a rectangle */}
/>
```

# We Need Your Feedback

## Live Sandbox

You can check out the [prototype](https://codesandbox.io/s/qk0zwwx929).

A package that contains the described functionality has the `next` tag:

```
npm i --save @devexpress/dx-react-core@next @devexpress/dx-react-chart@next
npm i --save @devexpress/dx-react-chart-material-ui@next
OR
npm i --save @devexpress/dx-react-chart-bootstrap4@next
```

[Documentation](https://devexpress.github.io/devextreme-reactive/@next/react/chart/docs/guides/zoom-and-pan/).

## Take a Quick Poll

[Do you find zooming and panning in the Chart useful?](https://docs.google.com/forms/d/e/1FAIpQLSdK8-XDbZO57DGyyPa9cY-Tarrgr2Q3RzT2p_84nDtxDrTcGw/viewform)

# Get Notified of Updates

Subscribe to this thread or to our [Facebook](https://www.facebook.com/DevExpress.DevExtreme/) and [Twitter](https://twitter.com/devextreme) accounts for updates on this topic.

**Comment by MaximKudriavtsev** [comment]
We have released the `@next` version of [DevExtreme Reactive Chart](https://devexpress.github.io/devextreme-reactive/@next/react/chart/docs/guides/getting-started/) with this feature. Feel free to provide your feedback about its functionality and [documentation](https://devexpress.github.io/devextreme-reactive/@next/react/chart/docs/guides/zoom-and-pan/).

A package that contains the described functionality has the `next` tag:

```
npm i --save @devexpress/dx-react-core@next @devexpress/dx-react-chart@next
npm i --save @devexpress/dx-react-chart-material-ui@next
OR
npm i --save @devexpress/dx-react-chart-bootstrap4@next
```

**Comment by MaximKudriavtsev** [comment]
We have released the `1.11.0` version of [DevExtreme Reactive Chart](https://devexpress.github.io/devextreme-reactive/react/chart/docs/guides/zoom-and-pan/) with this feature.

A package that contains the described functionality has the `latest` tag:

```
npm i --save @devexpress/dx-react-core @devexpress/dx-react-chart
npm i --save @devexpress/dx-react-chart-material-ui
OR
npm i --save @devexpress/dx-react-chart-bootstrap4
```
